### ISSUE TITLE
Cleanup odo create validation headers and spinners for devfile

### ISSUE THREAD

**Issue by maysunfaisal** [openshift/odo#3924]
Looks like `odo create` does two validations - 
- ValidateK8sResourceName in Validate()
- ParseFromDataAndValidate in Run()->devfileRun()

So if there is an actual devfile validation error when validating the components/commands/events in ParseFromDataAndValidate, it says validation passed from ValidateK8sResourceName and then outputs the ParseFromDataAndValidate error:
```
Maysuns-MacBook-Pro:springboot-ex maysun$ odo create myspr
Experimental mode is enabled, use at your own risk

Validation
 ✓  Creating a devfile component from devfile path: /Users/maysun/dev/redhat/resources/springboot-ex/devfile.yaml [51615ns]
 ✓  Validating devfile component [203508ns]
 ✗  the command "devbuild" does not map to a container component
```

We should be precise with the headers and spinners

/kind cleanup
/area devfile

**Comment by johnmcollier** [comment]
As a side note, we should probably also show more than _just_ the validation header. For example, `  ✓  Creating a devfile component from devfile path` really doesn't belong under `Validation`

**Comment by girishramnani** [comment]
we need to add spinner for storage related operations too as it was present for s2i components

**Comment by niconosenzo** [comment]
I would like to have a look at this one if you don't mind :).

**Comment by dharmit** [comment]
Closed by #4325

/close
